spec/test_typedstruct.rb

```diff
@@ -0,0 +1,567 @@
+#!/usr/bin/env -S bundle exec ruby
+# -*- coding: us-ascii -*-
+# frozen_string_literal: false
+require 'test/unit'
+require 'timeout'
+require 'typed_struct'
+
+##
+# This file is copied from the ruby test suite and modified for this gem:
+#   https://github.com/ruby/ruby/blob/master/test/ruby/test_struct.rb
+# Associated helper files under /vendor directory are from:
+#   https://github.com/ruby/ruby/tree/master/tool/lib
```

Comment:
Thanks for the update! I have to say though, I don't love having these copied over from another code base, especially with everything else from the test helpers - though I understand how it could be helpful, so I'm not sure what the best solution is here.

Looking at the tests themselves, a lot of them seem to be testing some of Struct's basic methods that this gem doesn't [deliberately] touch.

Just thinking out loud here, but perhaps something like this could be a simple option of comparing functionality?

```ruby
RSpec.describe TypedStruct do
  # maybe it could be helpful to more complexity here, but not sure
  let!(:struct) { Struct.new(:a, :b, :c) }
  let!(:typed_struct) { TypedStruct.new(a: Rbs("untyped"), b: Rbs("untyped"), c: Rbs("untyped")) }

  it "has the same singleton methods as a normal struct" do
    struct.methods.each do |method|
      expect(struct.send(method)).to eq typed_struct.send(method)
    end
  end

  it "has the same behaviour as a normal struct" do
    str = struct.new(1, 2, 3)
    ts = typed_struct.new(a: 1, b: 2, c: 3)

    str.methods.each do |method|
      expect(str.send(method)).to eq ts.send(method)
    end
  end
end
```

Though obviously not everything would have zero arity and there may be some methods which we clearly be different eg. `class` 🤷🏻 `(struct.methods - Object.methods).select { |m| struct.method(m).arity.zero? }`

Reply:
But in addition, I wouldn't mind so much if compatibility is _completely_ exact just at the moment (as long as it is always trending upwards over time), given the relatively early stage of this gem (pre 1.0)